apps/glusterfs/app.go

```diff
@@ -83,6 +83,15 @@ func NewApp(configIo io.Reader) *App {
 	// Setup asynchronous manager
 	app.asyncManager = rest.NewAsyncHttpManager(ASYNC_ROUTE)
 
+	switch app.conf.Allocator {
+	case "":
+		app.conf.Allocator = "simple"
+	case "simple":
+	default:
+		panic(errors.New("Invalid allocator specified: " + app.conf.Allocator))
+	}
+	logger.Info("Using %v allocator", app.conf.Allocator)
+
```

Comment:
This is the only patch in this series that I have any quibbles with. I am thinking that we should be moving away from having the app care about the allocator at all. I agree with the idea in your comment of decoupling the Allocator function from the app but I'd go further and just drop the function entirely and make the code that needs an allocator instance "get one". This should not be a problem now since there's only one allocator ATM, and we can just drop the config option entirely.

Reply:
If you think this is too much code churn at the moment a standalone `func GetAllocator() Allocator` would also be fine. FWIW I did a test where I removed the update of app.conf.Allocator and ran the unit tests. Everything passed; so this reassignment is probably superfluous and can be dropped.

Reply:
Sure, we don't currently need the config.
I just didn't want to do too big a removal in one step,  but if you are kind of sure that we will not need it any more, I'm more than happy to remove this altogether! Just let me know...

Reply:
removed this patch for now